src/Hyperion/Analysis.hs

```diff
@@ -49,11 +49,13 @@ namesOf = go []
     coerce = contramap (const ()) . fmap (const ())
 
 analyze
-  :: Text -- ^ Benchmark name
-  -> Sample -- ^ Measurements
+  :: Text -- ^ Executable name.
+  -> Text -- ^ Benchmarks name.
+  -> Text -- ^ Benchmark name.
+  -> Sample -- ^ Measurements.
   -> Report
-analyze name samp = Report
-    { _reportBenchName = name
+analyze executableName benchesName name samp = Report
+    { _reportBenchName = Text.concat [executableName, ".", benchesName, ".", name]
```

Comment:
That's not exactly what I remarked. Let's at least agree on some nomenclature here:

* Cabal package name = the name in the .cabal (passed manually by the user).
* Cabal benchmark name = the name of the benchmark executable (inferred via `getProgName`)
* Benchmark group = name passed to `bgroup`.
* benchmark name (or "case name" to disambiguate) = name passed to `bench`.

A project can have multiple Cabal packages, each one of which can have multiple Cabal benchmark names. Each Cabal benchmark can define multiple benchmark cases, optionally grouped together in some hierarchy internal to that benchmark suite.

A user should only only ever call `defaultMain` _once_ in any Cabal benchmark. That's why it's called `*Main` after all.